### DISCUSSION TITLE
Git Replacement libgit2?

### DISCUSSION THREAD

**Discussion by patgmiller** [gopasspw/gopass#2677] [Q&A] [answered]
I've read through the handful of issues considering replacement options for `git`, but i never found a reason why `libgit2` was not chosen or tried? If i had to guess a reason i think it would be related to the low level api nature of it, but I wanted to ask find out for sure.

here are the reference links that i read
- https://github.com/gopasspw/gopass/issues/155
- https://github.com/gopasspw/gopass/issues/1365
- https://github.com/gopasspw/gopass/issues/1704

**Comment by dominikschulz** [comment, marked as answer]
Because it's a C library: `libgit2 is a portable, pure C implementation of the Git core methods` [source](https://libgit2.org/).

We aim to support most platforms that Go supports and setting up CGo compatible builders for all of these is way beyond my abilities (mostly time related).

If anyone has a good, sustainable plan to do that, I'd be happy to listen. Otherwise anything that involves CGo is off-limits for this project.

**Reply by patgmiller** [reply]
Fair enough. I was trying to decide if i wanted to take a stab at either adding a `git` backend using the `libgit2`, or trying to implement the `merge` functionality in `go-git` and then add a backend with that. sound like the latter is the preferred approach Thanks @dominikschulz
